### PR TITLE
Add documentation for the state module

### PR DESCRIPTION
**Pull request by DidiBear**
This adds documentation to functions in the `state` module.

The goal is to make it easier for newcomers to understand what things are used for.

### CODE REVIEW

**Comment by happenslol** [comment]
Awesome, thanks for this! I'll drop in some minor grammar corrections if that's fine with you, but overall I'd leave it like this. I'll get this merged on the weekend.

**Comment by DidiBear** [comment]
Thanks! Feel free to correct those, I could have made mistakes as a non-native speaker :)

File: src/state.rs [hunk 1]
```diff
@@ -57,9 +69,11 @@ impl SimpleState for MyState {
     }
 }
 
+/// Creates in the `world` a camera entity.
+///
+/// The `dimensions` are used to center the camera in the middle
+/// of the screen and to make it covers the entire screen.
```
Comment (happenslol) [inline src/state.rs:75]:
```suggestion
/// Creates a camera entity in the `world`.
///
/// The `dimensions` are used to center the camera in the middle
/// of the screen, as well as make it cover the entire screen.
```

File: src/state.rs [hunk 2]
```diff
@@ -109,6 +127,8 @@ fn load_sprites(world: &mut World) -> Vec<SpriteRender> {
         .collect()
 }
 
+/// Creates in the `world` an entity for each of the provided `sprites`.
```
Comment (happenslol) [inline src/state.rs:130]:
```suggestion
/// Creates an entity in the `world` for each of the provided `sprites`.
```

**Pull request merged**